libslide/src/bignum/bignum_add.rs

```diff
@@ -0,0 +1,143 @@
+mod bignum_add {
+    use crate::bignum::Bignum;
+    use std::ops;
+
+    // @todo - figure out a swap method to handle lines 17-26
+    impl ops::Add for Bignum {
+        type Output = Bignum;
+
+        fn add(self, rhs: Bignum) -> Bignum {
+            // this assumes the signs are both positive for now
+            let mut carry: u8 = 0;
+            let mut lhs_size: usize = self.dec.iter().count();
+            let mut rhs_size: usize = rhs.dec.iter().count();
+            let mut res_dec: Vec<u8> = Vec::new();
+            let mut big_vec: Vec<u8>;
+
+            // HANDLE DECIMAL
+
+            // swap if neccessary to make lhs smaller. Initialize big vec as the bigger vector
+            if lhs_size > rhs_size {
+                let tmp: usize = lhs_size;
+                lhs_size = rhs_size;
+                rhs_size = tmp;
+                big_vec = self.dec.clone();
+            } else {
+                big_vec = rhs.dec.clone();
+            }
+
+            // compute the decimal addition up to the size of the small vec
+            for i in (0..lhs_size).rev() {
+                res_dec.push((self.dec[i] + rhs.dec[i] + carry) % 10);
```

Comment:
This is assuming self is the normalized `lhs`. What you should do is get a reference to both and swap the references if they should be:

```rust
let mut lhs_dec = &&self.dec; // why double reference? We want a pointer to the reference of the vector, so we can mutate what we're pointing at.
let mut rhs_dec = &&rhs.dec;

if (*lhs_dec).len() > (*rhs_dec).len() {
    std::mem::swap(&mut lhs_dec, &mut rhs_dec);
}
```

Reply:
why is this any better? you're calling .len() everytime you need the size.

Reply:
Oh I'm dumb. nevermind